### PR TITLE
Add docs on `run_from` fields

### PR DESCRIPTION
**Pull request by andrewvc**
## What does this PR do?

Fixes https://github.com/elastic/synthetics-service/issues/635

Documents the new `run_from` field, which was added in https://github.com/elastic/beats/pull/30632 .

This is a pure docs PR, since the implementation came with states.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: heartbeat/docs/heartbeat-observer-options.asciidoc [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
 == Add observer and geo metadata
 
-It is often useful to view and query various attributes of the instance Heartbeat is running on. This data is attached to events via the <<add-observer-metadata,`add_observer_metadata`>> processor. This processor populates the ECS `observer.*` fields. One field of particular utility is the `observer.geo.name` field, which you can configure via this processor. Use this field to create distinctive geographic regions for your uptime checks. You might label your regions by datacenter name or geographic region, e.g. `virginia-dc-1`, or `us-east-1a`, or `virginia-us`.
+Use the `heartbeat.run_from` option to set the geographic location fields relevant to a given heartbeat instance.
 
-Note that Heartbeat should not be used with the similarly named <<add-host-metadata,`add_host_metadata`>> processor. In ECS parlance the host is the thing that is monitored, not the thing doing the monitoring, which is the observer.
+Determines which geographic location to label this monitor as running from. Note, this
+can also be handled by setting the `run_from` option on an individual monitor to apply
+a unique setting to just that monitor.
```
Comment (colleenmcginnis) [inline heartbeat/docs/heartbeat-observer-options.asciidoc:8]: A couple thoughts:

* The first sentence here is unclear. What is doing the determining?
* The second sentence could be simplified.
* Is something like the suggestion below still accurate? Feel free to adjust as needed.

```suggestion
The `run_from` option is used to label the geographic location where the monitor is running.
Note, you can also set the `run_from` option on an individual monitor to apply a unique setting to just that monitor.
```

File: heartbeat/docs/heartbeat-observer-options.asciidoc [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
 
+Takes two top-level fields: `id` and `geo`, where `id` is a string used to uniquely identify
+the geographic location, and indexed as the `observer.name` field, and `geo` which is a map
+conforming to https://www.elastic.co/guide/en/ecs/current/ecs-geo.html[ECS geo fields] 
+and is indexed under `observer.geo`.
```
Comment (colleenmcginnis) [inline heartbeat/docs/heartbeat-observer-options.asciidoc:13]: An unordered list might be easier to read:

```suggestion
The `run_from` option takes two top-level fields:

* `id`: A string used to uniquely identify the geographic location. It is indexed as the `observer.name` field.
* `geo`: A map conforming to https://www.elastic.co/guide/en/ecs/current/ecs-geo.html[ECS geo fields]. It is indexed under `observer.geo`.
```

File: heartbeat/docs/monitors/monitor-common-options.asciidoc [hunk 3]
```diff
@@ -105,6 +105,46 @@ If the timeout is exceeded, {beatname_uc} publishes a `service-down` event. If t
...
+=== `run_from`
+
+Determines which geographic location to label this monitor as running from. Note, this
+is usually better handled by using the top level `heartbeat.run_from` option which
+takes the same values, but applies them to every monitor.
+
+Takes two top-level fields: `id` and `geo`, where `id` is a string used to uniquely identify
+the geographic location, and indexed as the `observer.name` field, and `geo` which is a map
+conforming to https://www.elastic.co/guide/en/ecs/current/ecs-geo.html[ECS geo fields]
+and is indexed under `observer.geo`.
```
Comment (colleenmcginnis) [inline heartbeat/docs/monitors/monitor-common-options.asciidoc:119]: Copy over the changes from above?

**Review by colleenmcginnis — COMMENTED** [review]
@andrewvc I left some thoughts below. Feel free to adjust my suggestions as needed!

**Comment by andrewvc** [comment]
@colleenmcginnis I've incorporated all your feedback, can you take another look?

File: heartbeat/docs/monitors/monitor-common-options.asciidoc [hunk 4]
```diff
@@ -105,6 +105,47 @@ If the timeout is exceeded, {beatname_uc} publishes a `service-down` event. If t
...
+
+Use the `run_from` option to set the geographic location fields relevant to a given heartbeat monitor.
+
+The `run_from` option is used to label the geographic location where the monitor is running.
+Note, you can also set the `run_from` option on all monitors via the `heartbeat.run_from` option.
+
+The `run_from` option takes two top-level fields:
+
+* `id`: A string used to uniquely identify the geographic location. It is indexed as the `observer.name` field.
+* `geo`: A map conforming to https://www.elastic.co/guide/en/ecs/current/ecs-geo.html[ECS geo fields]. It is indexed under `observer.geo`.
```
Comment (colleenmcginnis) [inline heartbeat/docs/monitors/monitor-common-options.asciidoc:120]:
```suggestion
* `geo`: A map conforming to {ecs-ref}/ecs-geo.html[ECS geo fields]. It is indexed under `observer.geo`.
```

File: heartbeat/docs/heartbeat-observer-options.asciidoc [hunk 5]
```diff
@@ -1,7 +1,27 @@
...
+Use the `heartbeat.run_from` option to set the geographic location fields relevant to a given heartbeat instance.
 
-Note that Heartbeat should not be used with the similarly named <<add-host-metadata,`add_host_metadata`>> processor. In ECS parlance the host is the thing that is monitored, not the thing doing the monitoring, which is the observer.
+The `run_from` option is used to label the geographic location where the monitor is running.
+Note, you can also set the `run_from` option on an individual monitor to apply a unique setting to just that monitor.
 
+The `run_from` option takes two top-level fields:
+
+* `id`: A string used to uniquely identify the geographic location. It is indexed as the `observer.name` field.
+* `geo`: A map conforming to https://www.elastic.co/guide/en/ecs/current/ecs-geo.html[ECS geo fields]. It is indexed under `observer.geo`.
```
Comment (colleenmcginnis) [inline heartbeat/docs/heartbeat-observer-options.asciidoc:12]:
```suggestion
* `geo`: A map conforming to {ecs-ref}/ecs-geo.html[ECS geo fields]. It is indexed under `observer.geo`.
```

**Review by colleenmcginnis — APPROVED** [review]
Two last suggestions -- we should use the [shared attribute](https://github.com/elastic/docs/blob/master/shared/attributes.asciidoc#L93) for the base path to the ECS docs.

**Pull request merged**